images/gcb-docker-gcloud/Dockerfile

```diff
@@ -12,9 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG GO_VERSION
+ARG GO_VERSION=1.22.9
```

Comment:
hmm, this change means you can accidentally build without the version we're actually building, which won't accurately reflect if the image build works (consider a breaking change in the go tagged images since this isn't just setting GOTOOLCHAIN or something like that)

Are we building this image with variants? We should probably pull this out of https://github.com/kubernetes/test-infra/blob/6574cf545fbe9646ede61e7a9e5d6be2d16e645b/images/gcb-docker-gcloud/cloudbuild.yaml#L17 and stop setting it in two places (IMHO just setting it here makes sense?)

Reply:
(it looks like we don't vary this input in the cloudbuilds -- no variants, only one value that just lives in the cloudbuild.yaml when it could live here instead and be more convenient)

Reply:
alight I can remove that, just that if someone building it from the `docker build` they have to use the build-arg always.

I will remove this one to keep the current behaviour as is.

Reply:
Ah I actually meant we should remove it from the cloud build and only define it here but we can do that in a follow-up

Reply:
I thought other way around, now I removed it from the cloudbuild.yaml file and have it only in the Dockerfile, ptal latest code.